Branch-SDK/src/main/java/io/branch/referral/Branch.java

```diff
@@ -1900,21 +1900,34 @@ private boolean pathMatch(String templatePath, String path) {
     }
 
     /**
-     * Enable Logging, independent of Debug Mode.
+     * Enable logging with a specific log level, independent of Debug Mode.
+     *
+     * @param iBranchLogging Optional interface to receive logging from the SDK.
+     * @param level The minimum log level for logging output.
+     */
+    public static void enableLogging(IBranchLoggingCallbacks iBranchLogging, BranchLogger.BranchLogLevel level) {
+        BranchLogger.setLoggerCallback(iBranchLogging);
+        BranchLogger.setMinimumLogLevel(level);
+        BranchLogger.setLoggingEnabled(true);
+        BranchLogger.logAlways(GOOGLE_VERSION_TAG);
+    }
+
+    /**
+     * Enable Logging, independent of Debug Mode. Defaults to DEBUG level.
      */
     public static void enableLogging() {
-        enableLogging(null);
+        enableLogging(null, BranchLogger.BranchLogLevel.DEBUG);
     }
 
     /**
-     * Optional interface. Implement a callback to receive logging from the SDK directly to your
+     * Enable Logging, independent of Debug Mode. Defaults to DEBUG level.
+     * Implement a callback to receive logging from the SDK directly to your
      * own logging solution. If null, and enabled, the default android.util.Log is used.
-     * @param iBranchLogging
+     *
+     * @param iBranchLogging Optional interface to receive logging from the SDK.
      */
-    public static void enableLogging(IBranchLoggingCallbacks iBranchLogging){
-        BranchLogger.setLoggerCallback(iBranchLogging);
-        BranchLogger.logAlways(GOOGLE_VERSION_TAG);
-        BranchLogger.setLoggingEnabled(true);
+    public static void enableLogging(IBranchLoggingCallbacks iBranchLogging) {
+        enableLogging(iBranchLogging, BranchLogger.BranchLogLevel.DEBUG);
```

Comment:
Does this mean customers using this feature will no longer produce logs for `VERBOSE` by default? This will break those workflows.

If it's not set, the original behavior should be maintained.

Reply:
This could also break our own automation tests.

Reply:
Updated it to set logs to Verbose when using a callback.